sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -508,71 +499,39 @@ private void pickOrCreateSample(PseudonymizableDto dto, LabMessageDto labMessage
+	private void editSample(SampleDto sample, LabMessageDto labMessage) {
 
-		Window window = VaadinUiUtil.createPopupWindow();
+		SampleController sampleController = ControllerProvider.getSampleController();
+		CommitDiscardWrapperComponent<SampleEditForm> sampleEditComponent = sampleController
+			.getSampleEditComponent(sample.getUuid(), sample.isPseudonymized(), FacadeProvider.getSampleFacade().getDiseaseOf(sample));
 
-		final CommitDiscardWrapperComponent<PathogenTestSelectionField> selectionField = new CommitDiscardWrapperComponent<>(selectField);
-		selectionField.getCommitButton().setCaption(I18nProperties.getCaption(Captions.actionConfirm));
-		selectionField.setWidth(1280, Sizeable.Unit.PIXELS);
-		selectionField.addCommitListener(() -> {
-			PathogenTestDto testDto = selectField.getValue();
-			if (testDto != null) {
-				editPathogenTest(sampleDto, testDto, caseSampleCount, labMessageDto.getTestedDisease(), labMessageDto);
-			} else {
-				createPathogenTest(sampleDto, labMessageDto);
-			}
-			window.close();
-		});
-		selectField.setSelectionChangeCallback((commitAllowed) -> selectionField.getCommitButton().setEnabled(commitAllowed));
-		selectionField.getCommitButton().setEnabled(false);
-		selectionField.addDiscardListener(window::close);
+		// add existing tests to edit component
+		int caseSampleCount = FacadeProvider.getSampleFacade().caseSampleCountOf(sample);
 
-		showFormWithLabMessage(labMessageDto, selectionField, window, I18nProperties.getString(Strings.headingPickOrCreatePathogenTest), false);
-	}
+		List<PathogenTestDto> existingTests = FacadeProvider.getPathogenTestFacade().getAllBySample(sample.toReference());
+		for (PathogenTestDto existingTest : existingTests) {
+			sampleController.addPathogenTestComponent(sampleEditComponent, existingTest, caseSampleCount, true);
+		}
 
-	private void editPathogenTest(SampleDto sampleDto, PathogenTestDto testDto, int caseSampleCount, Disease disease, LabMessageDto labMessageDto) {
-		BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest = (pathogenTestDto, callback) -> {
-			if (isValidPathogenTest(disease, pathogenTestDto)) {
-				if (pathogenTestDto.getTestResult() != sampleDto.getPathogenTestResult()) {
-					ControllerProvider.getSampleController()
-						.showChangePathogenTestResultWindow(
-							new CommitDiscardWrapperComponent<>(new SampleCreateForm(disease)),
-							sampleDto.getUuid(),
-							pathogenTestDto.getTestResult(),
-							testedResult -> callback.run());
-				} else {
-					callback.run();
-				}
-			} else {
-				callback.run();
-			}
-			finishProcessingLabMessage(labMessageDto, pathogenTestDto);
-		};
+		// add newly submitted tests to sample edit component
+		List<String> existingTestExternalIds = existingTests.stream().map(t -> t.getExternalId()).collect(Collectors.toList());
+		while (existingTestExternalIds.remove(null)); // otherwise tests without external id will seen as match
```

Comment:
you could filter the stream instead of this one liner that is hard to understand